Assets/!/Scripts/Player/PlayerHandsController.cs

```diff
@@ -2,38 +2,24 @@
 
 public class PlayerHandsController : MonoBehaviour
 {
-    private bool _facingRight = true;
-
-    public void UpdatePosition(Vector2 pointerPosition)
+    private float angle;
+    public void UpdatePositionMouse(Vector2 pointerPosition)
     {
         Vector3 objectPos = Camera.main.WorldToScreenPoint(transform.position);
         pointerPosition.x = pointerPosition.x - objectPos.x;
         pointerPosition.y = pointerPosition.y - objectPos.y;
 
-        if (pointerPosition.x < 0 && transform.localScale.x > 0)
-        {
-            _facingRight = false;
-            // flip if we are facing right but mouse is on the left
-            Vector3 theScale = transform.localScale;
-            theScale.x *= -1;
-            transform.localScale = theScale;
-        }
-        else if (pointerPosition.x > 0 && transform.localScale.x < 0)
-        {
-            _facingRight = true;
-            // flip if mouse is on the right and we are facing left
-            Vector3 theScale = transform.localScale;
-            theScale.x *= -1;
-            transform.localScale = theScale;
-        }
-
-        float angle = Mathf.Atan2(pointerPosition.y, pointerPosition.x) * Mathf.Rad2Deg;
-
-        if (_facingRight)
-            angle = Mathf.Atan2(pointerPosition.y, pointerPosition.x) * Mathf.Rad2Deg;
-        else
-            angle = Mathf.Atan2(pointerPosition.y * -1, pointerPosition.x * -1) * Mathf.Rad2Deg;
-
+        transform.localScale = pointerPosition.x > 0 ? Vector3.one : new Vector3(1, -1, -1);
+        angle = Mathf.Atan2(pointerPosition.y, pointerPosition.x) * Mathf.Rad2Deg;
         transform.rotation = Quaternion.Euler(new Vector3(0, 0, angle));
     }
+
+    public void UpdatePositionGamepad(Vector2 pointerPosition)
+    {
+        //this keeps the position of the weapon at the last point before rotation stopped instead of resetting to the default position
+        if (pointerPosition == Vector2.zero) return;
```

Comment:
Out of curiosity, would this work with the mouse pointer? I'm just thinking if it works for both then you don't need the if gamepad do this bullshit

Reply:
I needed to create UpdatePositionGamepad inside the PlayerHandsController because the UpdatePositionMouse method has those first 3 lines at the top that somehow mess up the gamepad controlls all together and the aiming doesn't work at all.  We could perhaps look at it together and figure out what the issue is and then maybe we could just have one method for the two input systems.